### PR TITLE
IBX-455: Added selected content from bookmarks to the top of list content items

### PR DESCRIPTION
**Pull request by mateuszdebinski**
| Question      | Answer
| ------------- | ---
| Tickets       | [IBX-455](https://issues.ibexa.co/browse/IBX-455)
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/lib/UniversalDiscovery/UniversalDiscoveryProvider.php [hunk 1]
```diff
@@ -112,16 +111,40 @@ public function getColumns(
             $columnLocations = $locationPath;
         }
 
+        if ($locationPathCount > 0) {
+            $columnLocationIdBefore = (int) $locationPath[$locationPathLast - 1];
```
Comment (barw4) [inline src/lib/UniversalDiscovery/UniversalDiscoveryProvider.php:115]: the ternary operator could be used here

File: src/lib/UniversalDiscovery/UniversalDiscoveryProvider.php [hunk 2]
```diff
@@ -112,16 +111,37 @@ public function getColumns(
             $columnLocations = $locationPath;
         }
 
+        $locationPathIndex = $locationPathCount > 0 ? $locationPathLast - 1 : 0;
+        $columnLocationIdBefore = (int) $locationPath[$locationPathIndex];
```
[the remark this answers is not in the document]
Comment (mateuszdebinski) [reply src/lib/UniversalDiscovery/UniversalDiscoveryProvider.php:115]: This is not initial Column but last, I change this variable name to `$lastColumnLocationId`

File: src/lib/UniversalDiscovery/UniversalDiscoveryProvider.php [hunk 3]
```diff
@@ -112,16 +111,37 @@ public function getColumns(
...
         foreach ($columnLocations as $columnLocationId) {
             $columnLocationId = (int)$columnLocationId;
             $columnLocation = ($columnLocationId !== self::ROOT_LOCATION_ID)
                 ? $this->getRestFormat($this->locationService->loadLocation($columnLocationId))
                 : null;
 
+            $subItems = $this->getSubitemLocations($columnLocationId, 0, $limit, $sortClause);
+            $locations = $subItems['locations'];
+
+            $index = array_search($locationId, array_map(static function ($location) {
```
[the remark this answers is not in the document]
Comment (mateuszdebinski) [reply src/lib/UniversalDiscovery/UniversalDiscoveryProvider.php:127]: This is not Location, it's array

**Review by webhdx — APPROVED** [review]

**Review by konradoboza — APPROVED** [review]

**Review by ViniTou — COMMENTED** [review]
I think we should have all that new code wrapped into private method that describes whats going on here. As for now i see search, some shifts and I still dont know why (aside from PR title.).

**Review by damianz5 — APPROVED** [review]

**Review by Steveb-p — CHANGES_REQUESTED** [review]
> I think we should have all that new code wrapped into private method that describes whats going on here. As for now i see search, some shifts and I still dont know why (aside from PR title.).

I second to that. It's confusing enough that it shouldn't be merged in this form.

**Review by ciastektk — APPROVED** [review]

**Review by alongosz — APPROVED** [review]

**Review by mikadamczyk — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Review by bogusez — APPROVED** [review]

**Comment by lserwatka** [comment]
You can merge it up now

**Pull request merged**